lib/data/entity.js

```diff
@@ -87,6 +87,26 @@ const parseSubmissionXml = (entityFields, xml) => new Promise((resolve, reject)
   });
 });
 
+////////////////////////////////////////////////////////////////////////////
+// PARSING JSON FOR UPDATING ENTITIES
+
+const parseUpdateJson = (entity, properties, body) => {
+  const { data: newData } = body;
+
+  const { data } = { ...entity.aux.currentVersion };
```

Comment:
~~this can be simplified:
`const data = entity.aux.currentVersion.data`~~

Oh you are cloning here. Can you please use explicit clone function of ramdajs or add a comment